File: 2112-minimum-difference-between-highest-and-lowest-of-k-scores/2112-minimum-difference-between-highest-and-lowest-of-k-scores.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var minimumDifference = function(nums, k) {
    if(k==1) return 0;

    nums.sort((a,b) => a-b);
    let min = nums[nums.length-1];
    for(let i=0;i<=nums.length-k;i++){
        let diff =nums[i+k-1]-nums[i]
        min = Math.min(min,diff)
    }
    return min;
};

File: 2112-minimum-difference-between-highest-and-lowest-of-k-scores/2112-minimum-difference-between-highest-and-lowest-of-k-scores.cpp
class Solution {
public:
    int minimumDifference(vector<int>& nums, int k) {
        int n=nums.size();
        if(n==1){
            return 0;
        }

        sort(nums.begin(),nums.end());

        int min = nums[nums.size()-1];
        for(int i=0;i+k-1<nums.size();i++){
            int j=i+k-1;
            int diff = nums[j]-nums[i];
            min = min<diff ? min:diff;
        }
        return min;
    }
};